### PR TITLE
Update dependencies + analyzer changes

### PR DESCRIPTION
**Pull request by Zazo032**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

This rolls minimum Dart SDK to the bundled with Flutter's latest stable version. Also, bumps `pedantic` and `octo_image` to latest versions. The update of `octo_image` is required for placeholders to not be wrongly placed.

### :boom: Does this PR introduce a breaking change?

From `octo_image`:
```
[0.3.0] - 2020-09-07 
Fixed bug when fitting placeholders or error widgets.
BREAKING: Doesn't automatically center placeholders and error widgets anymore.
```

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/Baseflow/flutter_cached_network_image/blob/develop/CONTRIBUTING.md))
- [x] Relevant documentation was updated
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by renefloor** [comment]
Which analyzer change do you think helps with #463?

**Comment by Zazo032** [comment]
As pub analyzer may complain that `pedantic` was not using the latest available version, I updated it and fixed the new warnings. I removed the rules from `analysis_options.yaml` because I think they're already included by including `pedantic` rules (if I removed one that was not included in `pedantic`, please tell me). Also, added https://github.com/Baseflow/flutter_cached_network_image/pull/464/commits/5961d95fd8c3801f9843a209c90ad89c80639b32#diff-d62227308ce5fdb544ee6b0a8b157a69R18 which may help with pub's error as per https://github.com/flutter/plugins/pull/2658

**Comment by renefloor** [comment]
Thanks for explaining all you did!

File: lib/src/cached_image_widget.dart [hunk 1]
```diff
@@ -5,7 +5,6 @@ import 'package:flutter/widgets.dart';
 import 'package:flutter_cache_manager/flutter_cache_manager.dart';
 import 'package:octo_image/octo_image.dart';
 
-typedef F = void Function();
```
Comment (renefloor) [inline lib/src/cached_image_widget.dart]: @Zazo032 this one is unused right?

Comment (Zazo032) [reply lib/src/cached_image_widget.dart]: Right, missed it

**Review by renefloor — APPROVED** [review]

**Pull request merged**